src/Microsoft.Extensions.Primitives/StringValues.cs

```diff
@@ -11,24 +12,62 @@ namespace Microsoft.Extensions.Primitives
     /// <summary>
     /// Represents zero/null, one, or many strings in an efficient way.
     /// </summary>
-    public struct StringValues : IList<string>, IReadOnlyList<string>, IEquatable<StringValues>, IEquatable<string>, IEquatable<string[]>
+    public class StringValues : IList<string>, IReadOnlyList<string>, IEquatable<StringValues>, IEquatable<string>, IEquatable<string[]>
```

Comment:
From testing though we see that the class implementation was better in terms of performance. It is likely that we are copying this type around very frequently and therefore incurring too much cost copying the fields. I also don't see a point in having a default constructor for the class. There's not much point in creating an instance will all fields as null instead of just setting the instance to null.